addons/web/static/src/core/datepicker/datepicker.js

```diff
@@ -8,6 +8,7 @@ const { Component, hooks } = owl;
 const { DateTime } = luxon;
 const { useExternalListener, useRef, useState } = hooks;
 
+const now = DateTime.local();
```

Comment:
I'd move this next to `isValidStaticFormat`, and maybe name it something else than `now` (it is indeed _now_ when you create it, but it's no longer _now_ later on (weird to save a "now" value).